app/controllers/form_controller.rb

```diff
@@ -293,10 +304,22 @@ def remove_fixed_duplicate_set_ids(log)
     SalesLog.find_by(duplicate_set_id:)&.update!(duplicate_set_id: nil) if log.sales? && SalesLog.where(duplicate_set_id:).count == 1
   end
 
-  def new_duplicate_set_id
-    loop do
-      duplicate_set_id = SecureRandom.random_number(1_000_000)
-      return duplicate_set_id unless LettingsLog.exists?(duplicate_set_id:)
+  def new_duplicate_set_id(log)
+    if log.lettings?
+      LettingsLog.maximum(:duplicate_set_id).to_i + 1
```

Comment:
is it possible all lettings logs will have nil `duplicate_set_id` at some point? If so we should set a default value I think

Also, just checking stats are aware that a duplicate set id won't be for a unique set but can be reassigned to a different set later on?

Reply:
ah ignore the first bit, forgot `to_i` returns 0 on nil

Reply:
Yup! I thought this would break when I ran the tests, but all good!